packages/jobs/lib/runner/local.ts

```diff
@@ -0,0 +1,81 @@
+import getPort, { portNumbers } from 'get-port';
+import type { NodeProvider } from '@nangohq/fleet';
+import { execSync, spawn } from 'child_process';
+import { logger } from '../logger.js';
+import { envs } from '../env.js';
+import { Err, Ok, stringifyError } from '@nangohq/utils';
+import { getProvidersUrl } from '@nangohq/shared';
+
+const localRunnerPids = new Map<number, number>(); // Mapping Node.id to process PID
+
+export const localNodeProvider: NodeProvider = {
+    start: async (node) => {
+        try {
+            // Random port to avoid conflicts with other fleet runners accross local execution
+            const rndPort = Math.floor(Math.random() * 10000) + 10000;
+            const port = await getPort({ port: portNumbers(rndPort, rndPort + 100) });
```

Comment:
The main change (other than the pids map) is the random port since runners lives in the db across execution. Always assigning the same port would lead to runners across execution to have the same url

Reply:
Could we just use the socket native port handling to pick an unused port?

Reply:
Yeah, I was more just saying anything like that by default has a race condition in the OS where you're picking an unused port but it can be grabbed before you start the process using the port. Any TCP based service usually has an option where you start it without giving it a port and the OS will give you one that's unused and guarantee there's no race condition.

Reply:
the race condition you mention exists. I think they even mention it in the readme of the getPort npm package. Not worth spending too much time on this imho given how unlikely it is to happen and that it will be fixed next time it is run